shell/core/types.ts

```diff
@@ -73,6 +96,31 @@ export interface IPlugin {
   addRoute(route: RouteConfig): void;
   addRoute(parent: string, route: RouteConfig): void;
 
+  /**
+   * Adds an action/button to the UI
+   */
+   addAction(locationConfig: LocationConfig, action: IAction): void;
```

Comment:
We might want to pull out the where and when as separate args.

I think this makes it slightly easier for the typing.

We can declare something like:

```
export enum BuiltinActionPositions {
  HEADER: 'header'
};
```
then we can use overloads for `addTab`:

```
addAction(where: BuiltinActionPositions, when: Location, action: IAction): void;
addAction(where: string, when: Location, action: IAction): void;
```

Which will give us nice intellisense autocomplete against a set of built-in positions using the enum plus the ability to specify a custom string

Reply:
Done. "Opened" up the methods so that they accept a string as a new param:
https://github.com/rancher/dashboard/pull/7777/commits/b29a798ca3b73c12ed6255a5af1c1de4a9b67def